plugins/env/app/models/environment_variable_group.rb

```diff
@@ -0,0 +1,18 @@
+class EnvironmentVariableGroup < ActiveRecord::Base
+  default_scope -> { order(:name) }
+
+  has_many :environment_variables
+  has_many :stage_environment_variable_groups
+  has_many :stages, through: :stage_environment_variable_groups
+
+  validates :name, presence: true
+  accepts_nested_attributes_for :environment_variables, allow_destroy: true
+
+  def self.to_hash(stage, deploy_group_id)
+    stage.environment_variable_groups.each_with_object({}) do |ev_group, all|
+      ev_group.environment_variables.select { |ev| ev.deploy_group_id == deploy_group_id }.each do |ev|
+        all[ev.name] = ev.value
```

Comment:
the deploy_group specific settings are merged into the 'all' settings to
not have to build them multiple times

On Thu, May 28, 2015 at 6:17 AM, Shane Hender <notifications@github.com>
wrote:

> In plugins/env/app/models/environment_variable_group.rb
> <https://github.com/zendesk/samson/pull/397#discussion_r31231311>:
>
> > @@ -0,0 +1,18 @@
> > +class EnvironmentVariableGroup < ActiveRecord::Base
> > +  default_scope -> { order(:name) }
> > +
> > +  has_many :environment_variables
> > +  has_many :stage_environment_variable_groups
> > +  has_many :stages, through: :stage_environment_variable_groups
> > +
> > +  validates :name, presence: true
> > +  accepts_nested_attributes_for :environment_variables, allow_destroy: true
> > +
> > +  def self.to_hash(stage, deploy_group_id)
> > +    stage.environment_variable_groups.each_with_object({}) do |ev_group, all|
> > +      ev_group.environment_variables.select { |ev| ev.deploy_group_id == deploy_group_id }.each do |ev|
> > +        all[ev.name] = ev.value
>
> Are the 'all' scoped varables included previously to this hash?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/397/files#r31231311>.
>